src/eth_provider/starknet/kakarot_core.rs

```diff
@@ -142,12 +143,13 @@ pub fn to_starknet_transaction(
 
     let mut calldata = Vec::with_capacity(capacity);
     calldata.append(&mut vec![
-        FieldElement::ONE,        // call array length
-        *KAKAROT_ADDRESS,         // contract address
-        *ETH_SEND_TRANSACTION,    // selector
-        FieldElement::ZERO,       // data offset
-        signed_data.len().into(), // data length
-        signed_data.len().into(), // calldata length
+        FieldElement::ONE, // call array length
+        *KAKAROT_ADDRESS,  // contract address
+        // This is possible because the selector is hardcoded in the EOA contract
+        *ETH_SEND_TRANSACTION + retries.into(), // selector + retries
```

Comment:
can this ETH_SEND_TRANSACTION + retries.into()  overflow?

Reply:
not currently, because retries is limited to 255

Reply:
can't a selector be FELT::MAX - 100?

Reply:
right right but I mean I checked the selector value and it's < FELT::MAX - 100